sdk/core/Azure.Core/src/Shared/OperationInternalOfT.cs

```diff
@@ -223,27 +223,14 @@ public Response<T> WaitForCompletion(TimeSpan pollingInterval, CancellationToken
             return Response.FromValue(Value, rawResponse);
         }
 
-        /// <summary>
-        /// Sets the <see cref="OperationInternal{T}"/> state immediately.
-        /// </summary>
-        /// <param name="state">The <see cref="OperationState{T}"/> used to set <see cref="OperationInternalBase.HasCompleted"/> and other members.</param>
-        public void SetState(OperationState<T> state)
-        {
-            if (state.HasCompleted && state.HasSucceeded)
-            {
-                Value = state.Value!;
-            }
-            ApplyStateAsync(false, state.RawResponse, state.HasCompleted, state.HasSucceeded, state.OperationFailedException, throwIfFailed: false).EnsureCompleted();
-        }
-
-        protected override async ValueTask<Response> UpdateStateAsync(bool async, CancellationToken cancellationToken)
+        protected override async ValueTask<OperationState> UpdateStateAsync(bool async, CancellationToken cancellationToken)
         {
             OperationState<T> state = await _operation.UpdateStateAsync(async, cancellationToken).ConfigureAwait(false);
```

Comment:
This method on IOperation.UpdateStateAsync feels like it should be named GetStateAsync since it shouldn't be updating anything only this instance method should?